dace/frontend/python/newast.py

```diff
@@ -4188,9 +4215,26 @@ def visit_Call(self, node: ast.Call):
                     if mapping is not None:
                         # Two-step replacement (N -> __dacesym_N --> mapping[N])
                         # to avoid clashes
-                        symbolic.safe_replace(
-                            mapping,
-                            lambda m: sd.replace_properties_dict(newarr, m))
+                        for sym, symvalue in mapping.items():
```

Comment:
:( one of the reasons I refactored it is because some of the replacements don't actually work (e.g., when replacing something with a constant float value).
Is there a way to reuse the existing function and add an extra boolean argument?

Reply:
No, I can try to fix `replace_properties_dict` to ensure that `repl_dict` is indeed symbolic.